src/components/StringHelper.js

```diff
@@ -0,0 +1,136 @@
+import { ValueErrorException, TypeErrorException } from './BaseExceptions';
+
+class StringHelper {
+	static isEmpty( Str ) {
+		return Str.length === 0;
+	}
+
+	static __removeBracesFormattedString( Str ) {
+		return Str.substring( 1, Str.length - 1 );
+	}
+
+	static __readComplexFormattedString( Str ) {
+		let Match;
+		const FoundFormatIndices = [];
+		const NameExpression = /(\{[_$%&§!]*[a-zA-Z0-9]+([_$%&§!]*[a-zA-Z0-9]*)*\})|(\{\})/g;
+
+		do {
+			Match = NameExpression.exec( Str );
+			if ( Match !== null ) {
+				if ( Match[ 0 ] === '{}' ) {
+					throw new ValueErrorException(
+						'Cannot switch from automatic field numbering to manual field specification.'
+					);
+				}
+				// @last argument -> remove first and last index is remove braces
+				FoundFormatIndices.push( [ Match.index,
+					Match.index + Match[ 0 ].length,
+					StringHelper.__removeBracesFormattedString( Match[ 0 ] )
+				] );
+			}
+		}
+		while ( Match !== null );
+
+		return FoundFormatIndices;
+	}
+
+	static __readLookAHeadFormattedString( Str, LookAHead ) {
+		let UserDefinedIndex = '';
+		LookAHead++;
+
+		while ( Str.charAt( LookAHead ) > '/' && Str.charAt( LookAHead ) < ':' ) {
+			UserDefinedIndex += Str.charAt( LookAHead );
+			LookAHead++;
+		}
+
+		return [ LookAHead, UserDefinedIndex ];
+	}
+
+	static __readSimpelFormattedString( Str ) {
+		let Index, LookAHead, UserDefinedIndex;
+		const FoundFormatIndices = [];
+		let Manual = false;
+		let Counter = 0;
+
+		for ( Index = 0; Str.length > Index; Index++ ) {
+			if ( Str.charAt( Index ) === '{' ) {
+
+				UserDefinedIndex = StringHelper.__readLookAHeadFormattedString( Str, Index );
+				LookAHead = UserDefinedIndex[ 0 ];
+				UserDefinedIndex = UserDefinedIndex[ 1 ];
+
+				if ( Str.charAt( LookAHead ) !== '}' ) {
+					throw new ValueErrorException( `Expected token { got ${ Str.charAt( LookAHead ) }.` );
+				}
+
+				LookAHead++;
+
+				if ( StringHelper.isEmpty( UserDefinedIndex ) === false ) {
+					FoundFormatIndices.push( [ Index, LookAHead, parseInt( UserDefinedIndex ) ] );
+					Manual = true;
+				} else {
+					FoundFormatIndices.push( [ Index, LookAHead, Counter ] );
+					Counter++;
+				}
+
+				Index = LookAHead;
+			}
+		}
+
+		if ( Counter !== 0 && Manual === true ) {
+			throw new ValueErrorException( 'Cannot switch from automatic field numbering to manual field specification.' );
+		}
+
+		return FoundFormatIndices;
+	}
+
+	static format( Str ) {
```

Comment:
I'm really not a fan of this :expressionless:

Reply:
As we discussed this as a matter of maintenance - I suggests we take https://www.npmjs.com/package/string.format or (where ever it's possible) we rely on the nodejs utils implementation.

Reply:
https://www.npmjs.com/package/string.format seems to be neither widely used nor actively maintained. I'm not sure if that's an issue for us, but something to be aware of.

Either way, using a library for this is a good idea! The choice of library is up to you :)